app/controllers/api/automate_workspaces_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Api
+  class AutomateWorkspacesController < BaseController
+    def index
+      raise BadRequestError, "Fetches of collection is not allowed"
+    end
+
+    def show
+      obj = AutomateWorkspace.find_by(:guid => @req.c_id)
+      if obj.nil?
+        raise NotFoundError, "Invalid Workspace #{@req.c_id} specified"
+      end
+
+      render_resource(:automate_workspaces, obj)
+    end
+
+    def edit_resource(_type, id, data = {})
+      obj = AutomateWorkspace.find_by(:guid => id)
+      if obj.nil?
+        raise NotFoundError, "Invalid Workspace #{id} specified"
+      end
+      workspace = data["workspace"]
+      state_var = data["state_var"]
+      if workspace.blank? && state_var.blank?
+        raise BadRequestError, "No workspace or state_var specified for edit"
+      end
+      current_output = obj.output || {}
+      obj.output = current_output.deep_merge(data)
+      obj.save
+      obj
```

Comment:
None of this belongs in a controller.  The api controllers should be stupid, and the model should have all of this "how to patch an atuomate workspace" knowledge.

Reply:
👍 model/business logic stays in model.

Reply:
> The api controllers should be stupid

❤️ ❤️ ❤️